core/src/main/kotlin/org/evomaster/core/problem/graphql/builder/GraphQLActionBuilder.kt

```diff
@@ -153,10 +153,18 @@ object GraphQLActionBuilder {
         element: Table,
         type: GQMethodType,
         params: MutableList<Param>,
-        actionCluster: MutableMap<String, Action>
+        actionCluster: MutableMap<String, Action>,
+        state: TempState,
+
     ) {
-        val action = GraphQLAction(actionId, element.fieldName, type, params)
-        actionCluster[action.getName()] = action
+        if (state.inputTypeName[element.fieldName]?.isNotEmpty() == true) {
+            val action = GraphQLAction(actionId, state.inputTypeName[element.fieldName].toString(), type, params)
+            actionCluster[action.getName()] = action
+        } else {
+            val action = GraphQLAction(actionId, element.fieldName, type, params)
+            actionCluster[action.getName()] = action
```

Comment:
as `actionCluster[action.getName()] = action` is the same in both branches, shouldn't it be moved after the `if`?

Reply:
Done.